components/google_sheets/sources/new-row-added/new-row-added.js

```diff
@@ -144,21 +156,36 @@ module.exports = {
     ) {
       return;
     }
-    const { files, newPageToken } = await this.google_drive.getModifiedSheets(
+    const { file, newPageToken } = await this.google_drive.getModifiedSheet(
       pageToken,
       this.drive === "myDrive" ? null : this.drive,
-      sheetIDs
+      this.sheetID
     );
-    for (const file of files) {
+    if (file) {
```

Comment:
I'd move the setting of the `newPageToken` above this section so it's colocated with the code that retrieves it above. Then:

```javascript
if (!file) return;
```

so you can avoid the indentation of the rest of the code below.